### PR TITLE
i18n(zh-cn): translate `core-concepts/rendering-modes.mdx`

### PR DESCRIPTION
**Pull request by 100gle**
#### Description (required)

- translate `core-concepts/rendering-modes.mdx` to Chinese version.
- totally refactor `guides/server-side-rendering.mdx`

#### Related issues & labels (optional)

#4667

### CODE REVIEW

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 1]
```diff
@@ -2,80 +2,122 @@
 title: 服务端渲染
```
Comment (Genteure) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:2]:
```suggestion
title: 按需渲染适配器
```

Source:

```yml
title: On-demand Rendering Adapters
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 2]
```diff
@@ -2,80 +2,122 @@
...
+import RecipeLinks from '~/components/RecipeLinks.astro';
 
-服务器端渲染（SSR）是指按需在服务器上生成 HTML 页面并将其发送到客户端。
+Astro 允许你为部分或所有页面和端点选择**按需渲染**。这也被称为**服务器端渲染 (SSR)**：在服务器上根据请求生成 HTML 页面，并将其发送给客户端。一个**适配器**被用来在服务器上运行你的项目并处理这些请求。
 
-SSR 允许你：
+这种按需渲染允许你：
 
 - 在应用中实现登录状态的会话。
 - 从使用 `fetch` 动态调用的 API 呈现数据。
```
Comment (Genteure) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:13]:
```suggestion
- 从使用 `fetch()` 动态调用的 API 呈现数据。
```

**Review by Genteure — APPROVED** [review]
LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks for the amazing work 🙌

**Pull request merged**